query/sources/impl/src/main/java/org/codice/ddf/admin/sources/opensearch/persist/UpdateOpenSearchConfiguration.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.sources.opensearch.persist;
+
+import static org.codice.ddf.admin.common.report.message.DefaultMessages.failedPersistError;
+import static org.codice.ddf.admin.sources.services.OpenSearchServiceProperties.OPENSEARCH_FEATURE;
+import static org.codice.ddf.admin.sources.services.OpenSearchServiceProperties.openSearchConfigToServiceProps;
+
+import java.util.List;
+
+import org.codice.ddf.admin.api.DataType;
+import org.codice.ddf.admin.api.fields.FunctionField;
+import org.codice.ddf.admin.common.fields.base.BaseFunctionField;
+import org.codice.ddf.admin.common.fields.base.scalar.BooleanField;
+import org.codice.ddf.admin.common.services.ServiceCommons;
+import org.codice.ddf.admin.configurator.Configurator;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.configurator.OperationReport;
+import org.codice.ddf.admin.sources.fields.type.OpenSearchSourceConfigurationField;
+import org.codice.ddf.admin.sources.utils.SourceValidationUtils;
+import org.codice.ddf.internal.admin.configurator.actions.FeatureActions;
+import org.codice.ddf.internal.admin.configurator.actions.ManagedServiceActions;
+import org.codice.ddf.internal.admin.configurator.actions.ServiceActions;
+import org.codice.ddf.internal.admin.configurator.actions.ServiceReader;
+
+import com.google.common.collect.ImmutableList;
+
+public class UpdateOpenSearchConfiguration extends BaseFunctionField<BooleanField> {
+
+    public static final String FIELD_NAME = "updateOpenSearchSource";
+
+    public static final String DESCRIPTION =
+            "Updates an OpenSearch source configuration specified by the pid. Returns true on success and false on failure.";
```

Comment:
I think we should avoid stating what is returned back, if this fails there will be error messages returned in which case the result is ignored by servlet and set to null instead. We should do this for all these new functions.